lib/web/scripts/node-join/install.sh

```diff
@@ -758,87 +760,174 @@ if teleport_binaries_exist; then
-        # install rpm with package manager
-        ${PACKAGE_MANAGER_COMMAND} "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
-    # use rpm if we couldn't find a package manager
+        log_important "Can't figure out what Teleport format to use"
+        exit 1
+    fi
+}
+
+install_from_repo(){
+    if [[ "${REPO_CHANNEL}" == "" ]]; then
+        # By default, use the current version's channel.
+        REPO_CHANNEL=v"${TELEPORT_VERSION//.*/}"
+    fi
+    # shellcheck disable=SC1091
+    . /etc/os-release
+
+    if [ "$ID" == "debian" ] || [ "$ID" == "ubuntu" ]; then
+        # old versions of ubuntu require that keys get added by `apt-key add`, without
+        # adding the key apt shows a key signing error when installing teleport.
+        if [[
+            ($ID == "ubuntu" && $VERSION_ID == "16.04") || \
+            ($ID == "debian" && $VERSION_ID == "9" )
+        ]]; then
+            sudo apt install apt-transport-https gnupg -y
+            curl https://deb.releases.teleport.dev/teleport-pubkey.asc | sudo apt-key add -
+            echo "deb https://apt.releases.teleport.dev/${ID} ${VERSION_CODENAME} stable/${REPO_CHANNEL}" | sudo tee /etc/apt/sources.list.d/teleport.list
+        else
+            sudo curl https://deb.releases.teleport.dev/teleport-pubkey.asc \
+                -o /usr/share/keyrings/teleport-archive-keyring.asc
+            echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] \
+            https://apt.releases.teleport.dev/${ID} ${VERSION_CODENAME} stable/${REPO_CHANNEL}" | \
+            sudo tee /etc/apt/sources.list.d/teleport.list >/dev/null
+        fi
+        sudo apt-get update
+        sudo apt-get install -y ${TELEPORT_PACKAGE_NAME}
+    elif [ "$ID" = "amzn" ] || [ "$ID" = "rhel" ] || [ "$ID" = "centos" ] ; then
+        if [ "$ID" = "rhel" ]; then
+            VERSION_ID="${VERSION_ID//.*/}" # convert version numbers like '7.2' to only include the major version
+        fi
+        sudo yum install -y yum-utils
+        sudo yum-config-manager --add-repo \
+        "$(rpm --eval "https://yum.releases.teleport.dev/$ID/$VERSION_ID/Teleport/%{_arch}/stable/${REPO_CHANNEL}/teleport.repo")"
+        sudo yum install -y ${TELEPORT_PACKAGE_NAME}
     else
-        # install RPM (in upgrade mode)
-        log "Using rpm to install ${TEMP_DIR}/${DOWNLOAD_FILENAME}"
-        rpm -Uvh "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
+        echo "Unsupported distro: $ID"
+        exit 1
+    fi
+}
+
+is_repo_available(){
+    if [[ "${OSTYPE}" != "linux-gnu" ]]; then
+        return 1
     fi
+
+    # shellcheck disable=SC1091
+    . /etc/os-release
+
+    # Only the following distros are supported:
+    if [[ ! "$ID" =~ ^(ubuntu|debian|amzn|rhel|centos)$ ]]; then
+        return 1
+    fi
+
+    # Teleport's deb repo has the following Ubuntu Versions:
+    if [[ "$ID" == "ubuntu" && ! "$VERSION_ID" =~ ^(16\.04|18\.04|20\.04|22\.04) ]]; then
```

Comment:
It's hard to keep it up to date without manual effort.

We could always try to use the repo and in case of failure, fallback to the binary installation.
I think this would solve both issues (new repo channel, removed repo channel).

I'm a bit hesitant in doing that because it adds some complexity to a script that we have to manually test everytime.

Reply:
> We could always try to use the repo and in case of failure, fallback to the binary installation.
> I think this would solve both issues (new repo channel, removed repo channel).

Yeah, I was thinking about this too but as you said, idk how viable that is here.

If I was you I'd seek feedback from someone who's more up to date with our Linux repo stuff than me. 😶‍🌫️

Reply:
I'm ok with both options but prefer as is

Deferring to @camscale then :+1:

Reply:
I would recommend adding a comment [here](https://github.com/gravitational/teleport/blob/master/build.assets/tooling/cmd/build-os-package-repos/runners.go#L41) and [here](https://github.com/gravitational/teleport/blob/master/build.assets/tooling/cmd/build-os-package-repos/runners.go#L132) that states that this script must be updated when those maps are changed. I think that will help catch most of these issues.

Reply:
Done :+1: